apps/evm_family/evm_typed_data_helper.c

```diff
@@ -187,9 +193,17 @@ bool evm_get_typed_struct_data_digest(
          sizeof(ETH_SIGN_TYPED_DATA_IDENTIFIER) - 1);
 
   offset += sizeof(ETH_SIGN_TYPED_DATA_IDENTIFIER) - 1;
-  eip712_status = hash_struct(&(typed_data->domain), data + offset);
-  offset += HASH_SIZE;
-  eip712_status |= hash_struct(&(typed_data->message), data + offset);
+  if (0 < typed_data->domain.children_count) {
+    eip712_status = hash_struct(&(typed_data->domain), data + offset);
+    offset += HASH_SIZE;
+    if (0 < typed_data->message.children_count) {
+      eip712_status |= hash_struct(&(typed_data->message), data + offset);
+    }
+  } else if (!strncmp(UI_TEXT_EIP712_DOMAIN_TYPE,
```

Comment:
```suggestion
  } else if (0 == strncmp(UI_TEXT_EIP712_DOMAIN_TYPE,
```

this is clearer

Reply:
also, what is this `else...if` case? when & how could it pop up in a real-world scenario?

Reply:
for test case 1: [empty data ](https://github.com/trezor/trezor-firmware/blob/main/common/tests/fixtures/ethereum/sign_typed_data.json)

[https://github.com/trezor/trezor-firmware/blob/54468abd89af62f4a7395398eb358d5fe687430c/core/src/apps/ethereum/sign_typed_data.py#L93](# Setting the primary_type to "EIP712Domain" is technically in spec
# In this case, we ignore the "message" part and only use the "domain" part
# https://ethereum-magicians.org/t/eip-712-standards-clarification-primarytype-as-domaintype/3286)

Reply:
please add this as a comment.

Reply:
added